src/devices/multipleanalogsensorsserver/MultipleAnalogSensorsServer.cpp

```diff
@@ -245,31 +284,37 @@ bool MultipleAnalogSensorsServer::attachAll(const yarp::dev::PolyDriverList& p)
     if (!ok)
     {
         yError("MultipleAnalogSensorsServer: failure in opening port named %s.", m_streamingPortName.c_str());
-        detachAll();
+        close();
         return false;
     }
 
     ok = this->yarp().attachAsServer(m_rpcPort);
     if (!ok)
     {
         yError("MultipleAnalogSensorsServer: failure in attaching RPC port to thrift RPC interface.");
-        detachAll();
+        close();
         return false;
     }
 
     ok = m_rpcPort.open(m_RPCPortName);
     if (!ok)
     {
         yError("MultipleAnalogSensorsServer: failure in opening port named %s.", m_RPCPortName.c_str());
-        detachAll();
+        close();
         return false;
     }
 
     // Set rate period
     ok = this->setPeriod(m_periodInS);
     ok = ok && this->start();
+    if (!ok)
+    {
+        yError("MultipleAnalogSensorsServer: failure in starting thread.");
+        close();
+        return false;
+    }
 
-    return ok;
+    return true;
```

Comment:
I think this change is unnecessary

Reply:
Device finalization must follow every `ok` status check so that wrapped subdevices, ports and threads are correctly closed. This line pretends to assert that all previous usage of `ok`  has been handled and it's fine to proceed further regardless of its value.

Reply:
Yes sure, it is correct, but keeping `ok` as return value should be ok. Sorry for the pun :sweat_smile: